### DISCUSSION TITLE
Clipboard manager for xwayland and wayland compability

### DISCUSSION THREAD

**Discussion by wicaksonoleksono** [mylinuxforwork/dotfiles#438] [Q&A] [answered]
The defauilt clipshit doesnt work on wayland app and xwayland app. Is there any alternatives ? thanks !

**Comment by SSYH** [comment]
let me ask, a little more gently:

I am also having problems with clipboard (cut / copy / past / ...).
Usually, within a single browser (sp. chrome), I can C/P, but usually NOT when switching workspaces or apps.
Sometimes CTRL-SHIFT-C & CTRL-SHIFT-V works... but I haven't understood when it does, and when it doesn't.

How do I verify the right clipboard apps / utilities, etc. are configured / installed correctly?
Could this project beef up its C/C/P tests prior to release?

Thx - Sam

**Reply by wicaksonoleksono** [reply, marked as answer]
Some app runs on Xwayland and some app runs on Wayland natively, you can fix this by just making sure the app you are running runs on wayland instead of xwayland vice versa. there is a flag you can add to the .Desktop app, or you can run it under a wrapper. you can find this in its AUR repo for each application and search for server-display flags

**Reply by SSYH** [reply]
Wouldn't it be more convenient to take care of this automatically rather than having the user take some action?

**Reply by wicaksonoleksono** [reply]
Well, you chose to install Arch Linux, so what can you expect? This is a Wayland issue. In Wayland, you can't easily perform operations that span across different compositors because the architecture is designed to isolate clients for better security and efficiency. Each system has its drawbacks.

You can't fully automate this either, because the majority of users still rely on X11, so Wayland support doesn’t always come configured out of the box. However, you might consider creating a custom pacman hook to manage adding Wayland-specific flags to installed or updated applications. Just an idea—I’ve never tried it myself since I don’t use manually compiled operating systems.

and maybe you can contribute to this repo hahahah : )
